src/stateful-event-subscriber.ts

```diff
@@ -190,13 +276,19 @@ export abstract class StatefulEventSubscriber<State>
   //current state, then the current state will be updated and the invalid flag
   //can be reset.
   setState(state: DeepReadonly<State>, blockNumber: number): void {
+    if (this.dexHelper.config.isSlave) {
+      this.stateBlockNumber = blockNumber;
+      this.stateHistory[blockNumber] = state;
```

Comment:
I think this will be leaking memory because we aren't using the code below that clears old history.  It's not clear that we even need `stateHistory` for slave instances, so maybe it can be not created for those.

Reply:
I agree maybe removing state history make sense for slave instance